app/scripts/platforms/extension.js

```diff
@@ -66,6 +66,32 @@ class ExtensionPlatform {
     }
   }
 
+  queryTabs (cb) {
+    extension.tabs.query({}, tabs => {
+      cb(tabs)
+    })
+  }
+
+  currentTab () {
+    return new Promise(resolve => {
```

Comment:
Nit: It seems a bit inconsistent to return promises for `currentTab` and `closeTab`, but not the other platform APIs. This kinda hurts me to say because I'd _love_ for them all to return promises instead personally.

Maybe this can be viewed as the first step to using promises for all of them :man_shrugging:

Reply:
In the direction of making it a first step, I also changed `queryTabs` and `switchToTab` to use promises: c4c3c7022